app/src/main/java/com/example/trainingintive/viewmodels/MainViewModel.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.trainingintive.viewmodels
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.example.trainingintive.repository.Repository
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.schedulers.Schedulers
+
+class MainViewModel(
+    private val repository: Repository
+) : ViewModel() {
+
+    init {
+        repository.getDogImageUrl()
+            .subscribeOn(AndroidSchedulers.mainThread())
+            .observeOn(Schedulers.io())
+            .subscribe(
+                { _dog.postValue(it.message) },
+                { _dog.postValue("błąd") }
+            )
+    }
+    private val _dog = MutableLiveData<String>()
+    val dog: LiveData<String> = _dog
+}
+
+class MainViewModelFactory(
```

Comment:
Move it to separate file

Reply:
It was temporary, because now I have common factory for all viewmodels